app/src/main/java/com/mongodb/hellosyncrealm/ui/home/HomeViewModel.kt

```diff
@@ -1,108 +1,68 @@
 package com.mongodb.hellosyncrealm.ui.home
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Transformations
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.*
 import com.mongodb.hellosyncrealm.ui.home.model.VisitInfo
 import com.mongodb.hellosyncrealm.ui.home.model.updateCount
 import io.realm.Realm
 import io.realm.mongodb.App
 import io.realm.mongodb.Credentials
-import io.realm.mongodb.User
-import io.realm.mongodb.sync.SyncConfiguration
+import io.realm.mongodb.SyncConfiguration
+import io.realm.query
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 
 class HomeViewModel(private val realmApp: App) : ViewModel() {
 
     private val _text = MutableLiveData<String>().apply {
         value = "Welcome to Realm"
     }
-
-    private val _visitInfo = MutableLiveData<VisitInfo>()
-    val visitInfo: LiveData<Int> = Transformations.map(_visitInfo) {
-        it.visitCount
-    }
-
     val text: LiveData<String> = _text
 
-    private val _isLoading = MutableLiveData<Boolean>(false)
+    private val _isLoading = MutableLiveData(false)
     val isLoading: LiveData<Boolean> = _isLoading
 
+    val visitInfo: LiveData<Int> = Transformations.map(onRefreshCount().asLiveData()) {
+        it?.visitCount
+    }
+
     init {
         updateData()
     }
 
     private fun updateData() {
-        _isLoading.postValue(true)
-
-        fun onUserSuccess(user: User) {
-            val config = SyncConfiguration.Builder(user, user.id).build()
-
-            Realm.getInstanceAsync(config, object : Realm.Callback() {
-                override fun onSuccess(realm: Realm) {
-                    realm.executeTransactionAsync {
-                        var visitInfo = it.where(VisitInfo::class.java).findFirst()
-                        visitInfo = visitInfo?.updateCount() ?: VisitInfo().apply {
-                            partition = user.id
-                        }.updateCount()
-
-                        it.copyToRealmOrUpdate(visitInfo).apply {
-                            _visitInfo.postValue(it.copyFromRealm(this))
-                        }
-                        _isLoading.postValue(false)
-                    }
-                }
-
-                override fun onError(exception: Throwable) {
-                    super.onError(exception)
-                    //TODO: Implementation pending
-                    _isLoading.postValue(false)
-                }
-            })
-        }
-
-        realmApp.loginAsync(Credentials.anonymous()) {
-            if (it.isSuccess) {
-                onUserSuccess(it.get())
-            } else {
-                _isLoading.postValue(false)
+        viewModelScope.launch(Dispatchers.IO) {
+            _isLoading.postValue(true)
+            val user = realmApp.login(Credentials.anonymous())
```

Comment:
I don't think login again and building the entire `SyncConfiguration` is a good idea, IMO the SyncConf should be built once at startup then reused/injected whenever needed. Same for the Realm since it's now thread-safe to cache it/pass it across thread. 

Maybe you can use a repository pattern similar to https://github.com/realm/realm-kotlin-samples/blob/main/Bookshelf/shared/src/commonMain/kotlin/io/realm/sample/bookshelf/BookshelfRepository.kt